### PR TITLE
Simplify unique key API

### PR DESCRIPTION
**Pull request by dimagorelik**
Simplify specific implementations of unique key

### CODE REVIEW

File: integration-tests/src/test/java/com/kenshoo/pl/audit/commands/DeleteAuditedChild2Command.java [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 import com.kenshoo.pl.entity.Identifier;
-import com.kenshoo.pl.entity.SingleUniqueKeyValue;
+import com.kenshoo.pl.entity.SingleUniqueKey;
 import com.kenshoo.pl.entity.internal.audit.entitytypes.AuditedChild2Type;
 
 public class DeleteAuditedChild2Command extends DeleteEntityCommand<AuditedChild2Type, Identifier<AuditedChild2Type>> {
 
     public DeleteAuditedChild2Command(final long id) {
-        super(AuditedChild2Type.INSTANCE, new SingleUniqueKeyValue<>(AuditedChild2Type.ID, id));
+        super(AuditedChild2Type.INSTANCE, new SingleUniqueKey<>(AuditedChild2Type.ID).createIdentifier(id));
```
Comment (effiban) [inline integration-tests/src/test/java/com/kenshoo/pl/audit/commands/DeleteAuditedChild2Command.java:11]: For the final result, I'm not sure this is more convenient than the previous code. 
I would prefer not to use a constructor at all, maybe something like this instead:
```keyOf(AuditedChild2Type.ID).withValue(id))```

Comment (dimagorelik) [reply integration-tests/src/test/java/com/kenshoo/pl/audit/commands/DeleteAuditedChild2Command.java:11]: @effiban 
Let's speak about it

**Comment by dimagorelik** [comment]
@effiban 
Please, review all changes

File: integration-tests/src/test/java/com/kenshoo/pl/auto/inc/ParentEntity.java [hunk 2]
```diff
@@ -33,15 +30,15 @@ public SupportedChangeOperation getSupportedOperation()  {
...
-            super(ID, id);
-        }
-    }
-
     public static class IdInTarget extends SingleUniqueKeyValue<ParentEntity, Integer> {
+        public static final SingleUniqueKey<ParentEntity, Integer> DEFINITION = new SingleUniqueKey<>(ParentEntity.ID_IN_TARGET) {
+            protected SingleUniqueKeyValue<ParentEntity, Integer> createValue(Integer value) {
+                return new IdInTarget(value);
+            }
+        };
```
Comment (effiban) [inline integration-tests/src/test/java/com/kenshoo/pl/auto/inc/ParentEntity.java:38]: Why did you add this? Isn't this the older and more complex definition?

Comment (dimagorelik) [reply integration-tests/src/test/java/com/kenshoo/pl/auto/inc/ParentEntity.java:38]: @effiban 
I deleted constructor

**Review by effiban — APPROVED** [review]
